routing/pathfind.go

```diff
@@ -361,6 +369,15 @@ func edgeWeight(e *channeldb.ChannelEdgePolicy) float64 {
 	return float64(1 + e.TimeLockDelta)
 }
 
+// graphSource is a wrapper around a graph that can either be in memory or
+// on disk, like the regular channeldb.ChannelGraph.
+type graphSource struct {
+	ForEachNode    func(func(*channeldb.LightningNode) error) error
+	ForEachChannel func(*channeldb.LightningNode,
```

Comment:
Should this function just accept an `edge` struct instead of the three separate parameters?